plugins/woocommerce/src/Admin/API/AI/Product.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\API\AI;
+
+use Automattic\WooCommerce\Blocks\AIContent\UpdateProducts;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product controller
+ *
+ * @internal
+ */
+class Product extends AIEndpoint {
+	/**
+	 * The store title option name.
+	 *
+	 * @var string
+	 */
+	const AI_CONTENT_GENERATED = 'ai_content_generated';
+
+	/**
+	 * Endpoint.
+	 *
+	 * @var string
+	 */
+	protected $endpoint = 'product';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		$this->register(
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'update_product' ),
+					'permission_callback' => array( Middleware::class, 'is_authorized' ),
+					'args'                => array(
+						'products_information' => array(
+							'description' => __( 'Data generated by AI for updating dummy products.', 'woocommerce' ),
+							'type'        => 'object',
+						),
+						'last_product'         => array(
+							'description' => __( 'Whether the product being updated is the last one in the loop', 'woocommerce' ),
+							'type'        => 'boolean',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Update product with the content and image powered by AI.
+	 *
+	 * @param  WP_REST_Request $request Request object.
+	 *
+	 * @return WP_Error|WP_REST_Response
```

Comment:
I don't think this is accurate anymore, wont it always be `WP_REST_Response`?

Reply:
True, fixed! TY!